### DISCUSSION TITLE
Anyone using Herd + Horizon on Windows?

### DISCUSSION THREAD

**Discussion by wyxos** [beyondcode/herd-community#580] [Q&A] [answered]
I tried porting an existing project to Herd and while running composer update, I eventually came across the error
```
  Problem 1
    - laravel/horizon[v5.23.0, ..., v5.23.2] require ext-pcntl * -> it is missing from your system. Install or enable PHP's pcntl extension.
    - Root composer.json requires laravel/horizon ^5.23 -> satisfiable by laravel/horizon[v5.23.0, v5.23.1, v5.23.2].
```

I understand that this extension has no equivalent on Windows, ignoring the platform requirements, does not resolve the issue. 

Anyone has encountered this or is WSL the only solution in this case? 

This also makes me wonder if there are other package/similar cases of dependencies over PHP extensions that are not supported on Windows.

**Comment by GL-MorganC** [comment]
I didn't succeed to install Horizon on my Windows.. The only option I had is to install it on my project ignoring some requirements

`composer require --ignore-platform-req=ext-pcntl --ignore-platform-req=ext-posix laravel/horizon` 

You have to use those ignore parameters each time you are on windows and try to install /require any other packages.

**Comment by wyxos** [comment]
I understand the above solution enables installing the dependencies and skips the error, but it will not allow using Horizon still on Windows, right? 

Is there a way via Herd to do so? Otherwise I am trying to explore using WSL on Windows and keep Herd for Mac.

**Comment by GL-MorganC** [comment]
You 're right! This don't install Horizon on windows, it juste let update your composer.json with the package for your other environment.

Unless virtualisation, I did not see other solution to implement Horizon on windows :(

**Comment by TonsiTT** [comment]
[This guy](https://github.com/beyondcode/herd-community/issues/780) managed to run herd + horizon using WSL. Perhaps this could be looked into as a way for Herd team to support horizon since most devs are already familiar with WSL and wouldn't mind using.
 
 Also if anyone is doing this, would appreciate if you could gives steps

**Comment by sschlein** [comment, marked as answer]
Technically, that's not something that we can solve with Herd. I recommend doing a PR to the official Horizon docs at https://laravel.com/docs/11.x/horizon.
